README.md

```diff
@@ -134,23 +134,70 @@ interface "wasi:kv/data/set-many" {
   // Notice that this interface is non-atomic, meaning that the key-value pairs
   // may not be consistent.
   use { Error, key } from "wasi:kv/types"
-  
+
   set-many: func(key_values: stream<(key, stream<u8>)>) -> result<_, Error>
 }
 
+interface "wasi:kv/data/range" {
+  // request using a range of keys
+  use { Error, payload, key-range, key-value } from "wasi:kv/types"
+
+  // returns keys within the range
+  get-keys: func(range: key-range) -> result<stream<key>, Error>
+
+  // returns key-value tuples within the key range
+  get-range: func(range: key-range) -> Result<stream<key-value>, Error>
+
+  // returns the number of keys that match the range
+  count-range: func(range: key-range) -> result<u64, Error>
+
+  // deletes keys that match the range
+  delete-range: func(range: key-range) -> result<_, Error>
+}
+
 interface "wasi:kv/types" {
-  resource Error { 
+  resource Error {
     trace: func() -> string
 
     // possibly more methods
   }
+
   resource payload {
     consume_async: func() -> result<stream<u8>, Error>
     consume_sync: func() -> result<list<u8>, Error>
     size: func() -> result<u64, Error>
     // possibly more methods, like consume_async_with_key that returns a key payload pair `(key, stream<u8>)`
   }
 
+  // a combination of key and value
+  record key-value {
+    key: string,
+    value: payload,
+  }
+
+  // specification for a range of keys in a wasi:kv/data/range request
+  variant key-range {
+    // All keys
+    all,
+
+    // All keys beginning with the prefix.
+    // For example, "s" returns all keys staring with s
+    prefix(string),
+
+    // Inclusive is a closed interval including both endpoints.
+    // For example, in a range of timestamps, interval("2022-01-01T00:00:00", "2022-01-31T23:59:59")
+    // includes all days and times in January 2022.
+    // Caution: prefixes sort lexicographically before words containing the prefix, so the range
+    // `inclusive("a","z")` includes "allosaurus" and "yunnanosaurus", but not "zephyrosaurus".
```

Comment:
Thank you for providing these detailed comments, @stevelr ~ They are great!

Could you provide further clarification on this caution statement? Before reading, I had in mind that `inclusive("a", "z")` would return all words beginning with a-z while `interval("a", "z")` would return all words beginning with a-y. Why does `inclusive("a", "z")` not include "zephyrosaurus"?